java/arcs/core/data/expression/Expression.kt

```diff
@@ -254,6 +257,16 @@ sealed class Expression<T> {
         override fun <Result> accept(visitor: Visitor<Result>) = visitor.visit(this)
     }
 
+    /**
+     * The implicit scope used by the left most qualifier in a field lookup expression, e.g.
+     * `[num > 100]` in a refinement is actually `[currentScope.num > 100]` where
+     * `currentScope` would be bound during evaluation to an entity.
+     * @param T the type of the resulting lookup
+     */
+    class CurrentScopeExpression<T : Scope> : Expression<T>() {
```

Comment:
Would it be a bad idea to instead of a nullable qualifier in the FieldExpression?

Reply:
I guess you could treat 'null' as "look up in current scope", but it breaks the visitor pattern, because now there can't be a visit(CurrentScope) unless you put hack logic in visit(FieldExpression) on all visitors.